Fix N1QL creator to support @Field annotation on derived method queries.

mmonti opened the pull request:
This PR contains changes needed to have all the integration tests fixed. Some of these changes can be included in JHipster Generator and others are required in Spring Data Couchbase project and are described here: 
- https://github.com/jhipster/generator-jhipster/issues/13544#issuecomment-771007769

mmonti committed:
Fix N1QL creator to support @Field annotation

mikereiche commented:
https://github.com/spring-projects/spring-data-couchbase/issues/1055

mraible commented:
@mmonti What do you think the best way forward is? Should I integrate your changes in JHipster's main generator or should we wait for Spring Data Couchbase fixes? We could do another JHipster 7 beta without it since it's not working currently.

mmonti committed:
Removed QueryCriteria since is not used

mmonti commented:
I think we can do another JHipster 7 beta for now so at least we can have a version working. I will make another commit in this PR to leave only the files needed and you can integrate (we don't need the changes in QueryCriteria since I replaced "findAllByIdNotNullAndActivatedIsTrue" query with a @Query). I can also do it if needed.

I think that I misread the ticket that @mikereiche shared about [DATACOUCH-1055](https://github.com/spring-projects/spring-data-couchbase/pull/1056) and thought that change was merged, but seems that the ticket was just closed.

I did some tests with 4.2.0-SNAPSHOT and the classes I added to fix the current issues will not work (some of the classes I extended are deprecated in 4.2.x), so we will need to do another update when we decide to move from 4.1.3 to 4.2.x

mraible commented:
@mmonti OK. I'll integrate your changes into the main generator this evening.
